x-pack/plugins/enterprise_search/public/applications/app_search/components/source_engines/source_engines_logic.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { kea, MakeLogicType } from 'kea';
+
+import { Meta } from '../../../../../common/types';
+
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { HttpLogic } from '../../../shared/http';
+import { EngineLogic } from '../engine';
+import { EngineDetails } from '../engine/types';
+
+interface EnginesAPIResponse {
+  results: EngineDetails[];
+  meta: Meta;
+}
```

Comment:
[tech debt/optional] It looks like this is used in `engines_logic.ts` as well - I think it would be worth DRYing out to a `engines/types.ts` file for re-use! I can grab this in a small follow-up PR if you'd rather not do it here though, no worries

Reply:
I've addressed this in 0590e77

Reply:
Awesome!!! Thanks so much Byron 🎉 🎉